### PR TITLE
typing DeviceTypes attempt 2

### PR DESCRIPTION
**Pull request by AMarcedone**
Please take a look, specifically at the latest commit with the encoder. 
I guess enabling the decoders to decode both strings and ints is not strictly necessary as we are always encoding to string, but I figured it would be safer this way.

### CODE REVIEW

File: protocol/avdl/keybase1/common.avdl [hunk 1]
```diff
@@ -193,6 +193,18 @@ protocol Common {
     MOBILE_1
   }
 
+// these enum values need to match the ones in
+// keybase/lib_public/public_constants.ts ->
+// public_constants.device.type
+  enum DeviceTypeV2 {
```
Comment (maxtaco) [inline protocol/avdl/keybase1/common.avdl]: talked it over with @mmaxim, he's still nervous this is going to break something and be hard to debug. There's no way in a review I can be certain it won't break things.


let's stick with the original original plan:

```
@typedef ("string") record DeviceTypeV2 {}
```

Comment (maxtaco) [reply protocol/avdl/keybase1/common.avdl]: In general changing the AVDL in a way that's no backwards compatible is a huge can of worms and we never do it.

**Comment by AMarcedone** [comment]
ok @maxtaco this is ready for another look

File: go/protocol/keybase1/extras.go [hunk 2]
```diff
@@ -1876,6 +1880,36 @@ func PublicKeyV1FromDeviceKeyV2(keyV2 PublicKeyV2NaCl) PublicKey {
...
+	DeviceTypeV2_DESKTOP = "desktop"
+	DeviceTypeV2_MOBILE  = "mobile"
+)
+
+func (d DeviceTypeV2) String() string {
+	return string(d)
+}
+
+func StringToDeviceTypeV2(s string) (d DeviceTypeV2, err error) {
+	deviceTypeStr := UnquoteString(s)
```
Comment (maxtaco) [inline go/protocol/keybase1/extras.go]: why do you have to unquote here?

File: go/protocol/keybase1/extras.go [hunk 3]
```diff
@@ -1876,6 +1880,36 @@ func PublicKeyV1FromDeviceKeyV2(keyV2 PublicKeyV2NaCl) PublicKey {
 	}
 }
 
+const (
+	DeviceTypeV2_NONE    = "none"
```
Comment (maxtaco) [inline go/protocol/keybase1/extras.go]: shouldn't this be:

```
DeviceTypeV2_NONE DeviceTypeV2 = "none"
```

etc?

File: go/engine/passphrase_recover_test.go [hunk 4]
```diff
@@ -238,7 +238,7 @@ func TestPassphraseRecoverChangeWithPaper(t *testing.T) {
 		Password: "test1234",
 	}
 	loginUI = &TestLoginUIRecover{
-		chooseDevice: "paper",
```
Comment (maxtaco) [inline go/engine/passphrase_recover_test.go]: whooops that was a bug ^

File: go/protocol/keybase1/extras.go (hunk 2 above)
Comment (AMarcedone) [reply go/protocol/keybase1/extras.go]: leftover from before, got rid of it

**Pull request merged**